### ISSUE TITLE
Line "End beat" in status bar is incorrect

### ISSUE THREAD

**Issue by stevage** [musescore/MuseScore#34472]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Description with steps to reproduce

1. Create a line feature over a whole note, starting at the start of the bar.
2. Shorten it progressively down to half a beat in duration and observe the "End beat" in the status bar.

"End beat" is always reported as beat 1. Even if the line feature starts on beat 3, it will still be reported as ending on beat 1. (Similar but less extreme things happen when there are more, shorter notes, but I don't think I can exactly explain the logic.)

Now do the same over a whole note rest.

Now the "end beat" does update as you expand/contract the line feature, but it seems to be to be half a beat short. For example, if a feature begins at the start of the bar and lasts for 1 beat, it will be reported as "Start beat: 1 End beat: 1.5" (whereas I think it should say End beat: 2?)

### Supporting files, videos and screenshots

https://github.com/user-attachments/assets/0cf3ee8a-79ee-4a02-89aa-6a0fd22d89e1

<img width="638" height="289" alt="Image" src="https://github.com/user-attachments/assets/83306a49-6db0-4016-a4c9-9bc1f8cd45f8" />

### In which versions of MuseScore Studio is this issue present?

OS: macOS Sequoia (15.7.7), Arch.: arm64, MuseScore Studio version (64-bit): 4.7.4-260706075, revision: github-musescore-musescore-7688c00

### Regression

I was unable to check

### Operating system

macOS

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible
